### PR TITLE
Electrolyzer model

### PR DESCRIPTION
**Pull request by cameron-goddard**
### Summary

Implemented the electrolyzer model according to what we know about the electrolyzer setup on the spacecraft right now. There might be some changes later on depending on whether or not we will have two combustion chambers etc.

### Testing
Added some basic unit tests

### Notes
N/A

### Comments
- [x] Add an x between the brackets if you commented your code well!

### CODE REVIEW

**Comment by andrew032011** [comment]
As a hint for why the tests are failing, you updated the `Parameters` and `State` classes, but there are tests in `state_test.py` and `parameters_test.py` that need to be updated alongside that change, or else they will fail. In particular, if we look at the logs, `test_to_array()` in `state_test.py` simply checks that a State is being properly converted into an array, but our test case for a state_list is a list that doesn't incorporate the newly added fields for the State which you added, and neither does state_1 (from `test_utils.py`), so what happens is the fields of `state_1` are being set to their default values, and the test cases fails. 

Basically, you need to make some updates to `state_test.py`, `test_utils.py`, and `parameters_test.py` in order to "redefine" what the definition of a valid state/state-list and valid parameters are.

File: src/core/models/electrolyzer_model.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+      Returns:
+          Dict[str, Any]: The augmented fuel mass and chamber pressure
+      """
+
+      # get fuel mass at this time
+      fuel_mass_i = np.array([state_time.state.fuel_mass])
+      chamber_temp = np.array([state_time.state.chamber_temp])
+
+      # calculate updated fuel mass
+      print("duration:", self._duration)
```
Comment (adamrnasir) [inline src/core/models/electrolyzer_model.py:33]: Let's use logging rather than printing

File: tests/core_tests/model_tests/electrolyzer_model_test.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        electrolyzer_a = ElectrolyzerModel(10.0, param_a)
+        eval_a = electrolyzer_a.evaluate(state_a)
+
+        # verify objects exist/are reasonable
+        self.assertIsNotNone(eval_a)
+        self.assertIsInstance(electrolyzer_a, ElectrolyzerModel)
+        self.assertIsInstance(eval_a, Dict)
+        
+
+        print(eval_a)
```
Comment (adamrnasir) [inline tests/core_tests/model_tests/electrolyzer_model_test.py:28]: May want to remove this print statement for final version

**Review by adamrnasir — CHANGES_REQUESTED** [review]
Looks pretty good; just a couple items that need to be addressed that I added comments for. Additionally, it's currently not passing pytest because of tests that verify state fields— I think the addition of new fields needs to be reflected in those tests.

For the sake of consistency and traceability, I'll PR `event_based` first then we can PR this on top of that.

**Review by andrew032011 — APPROVED** [review]

**Review by adamrnasir — APPROVED** [review]
